core/metrics.py

```diff
@@ -21,179 +24,88 @@ def get_nodes_for_validator(skale, val_id):
-        done_len = int(round(bar_len * count / float(total)))
-
-        percents = round(100.0 * count / float(total), 1)
-        bar = '=' * done_len + '-' * (bar_len - done_len)
-
-        sys.stdout.write('[%s] %s%s %s\r' % (bar, percents, '%', status))
-        sys.stdout.flush()
-
-
-def get_start_end_block_numbers(skale, node_ids, start_date=None, end_date=None):
-    if start_date is None:
-        start_block_number = get_start_block(skale, node_ids[0])
-    else:
-        start_block_number = find_block_for_tx_stamp(skale, start_date)
-
-    if end_date is None:
-        last_block_number = skale.web3.eth.blockNumber
-    else:
-        last_block_number = find_block_for_tx_stamp(skale, end_date)
-
-    return start_block_number, last_block_number
+def get_metrics_for_validator(skale, val_id, start_date=None, end_date=None, wei=None,
+                              to_file=False):
+    class nodeThread(threading.Thread):
+        def __init__(self, node_id):
+            threading.Thread.__init__(self)
+            self.node_id = node_id
 
+        def run(self):
+            metrics = get_metrics_from_events(skale, self.node_id, start_date, end_date,
+                                              is_validator=True)
+            all_metrics.extend(metrics)
 
-def format_limit(limit):
-    if limit is None:
-        return float('inf')
+    node_ids = get_nodes_for_validator(skale, val_id, )
+    all_metrics = []
+    thread_list = []
+    for node_id in node_ids:
+        node_thread = nodeThread(node_id)
+
+        thread_list.append(node_thread)
+        node_thread.start()
+    for th in thread_list:
+        th.join()
+    if all_metrics:
+        columns = ['Date', 'Node ID', 'Bounty', 'Downtime', 'Latency']
+        df = pd.DataFrame(all_metrics, columns=columns)
+        if not wei:
+            df['Bounty'] = df['Bounty'].apply(to_skl)
+        df.sort_values(by=['Date'], inplace=True, ascending=False)
+        metrics_rows = df.values.tolist()
+        node_group = df.groupby(['Node ID'])
+        metrics_sums = node_group.agg({'Bounty': 'sum', 'Downtime': 'sum', 'Latency': 'mean'})
+        metrics_sums = metrics_sums.reset_index().values.tolist()
+        total_bounty = df['Bounty'].sum()
+        if to_file:
+            df.to_csv('metrics.csv', index=False)
     else:
-        return int(limit)
+        metrics_rows = metrics_sums = total_bounty = None
+    return {'rows': metrics_rows, 'totals': metrics_sums}, total_bounty
+
+
+def get_metrics_for_node(skale, node_id, start_date=None, end_date=None, wei=None, to_file=False):
+    metrics = get_metrics_from_events(skale, node_id, start_date, end_date)
+    columns = ['Date', 'Bounty', 'Downtime', 'Latency']
+    df = pd.DataFrame(metrics, columns=columns)
+    if not wei:
+        df['Bounty'] = df['Bounty'].apply(to_skl)
+    total_bounty = df['Bounty'].sum()
+    metrics_rows = df.values.tolist()
+    if to_file:
+        df.to_csv('node_metrics.csv', index=False)
```

Comment:
User should be able to set dest folder - add an option for this

Reply:
done